packages/platform-sdk-profiles/src/drivers/electron/repositories/profile-repository.ts

```diff
@@ -0,0 +1,126 @@
+import { injectable } from "inversify";
+import { ipcRenderer } from "electron";
+import { IProfile, IProfileExportOptions, IProfileRepository } from "../../../contracts";
+
+import { Profile } from "../profiles/profile";
+import { Exceptions } from "@arkecosystem/platform-sdk";
+
+@injectable()
+export class ProfileRepository implements IProfileRepository {
+
+	public constructor() {
+	}
+
+	public fill(profiles: object): void {
+		ipcRenderer.invoke("ProfileRepository.fill", profiles)
```

Comment:
This is where all the delegation happens. Whatever is invoked in this driver, it's forwarded to the ipc main, and then the response processed.
I'm trying to avoid adding async / await to the interfaces, as that would affect the in memory driver too, but I don't think I'll have a choice and I'll have to change it anyway.